CMakeLists.txt

```diff
@@ -57,161 +85,164 @@ if(APPLE)
+if(BUILD_SHARED_LIBS)
+	if(APPLE)
+		sentry_install(FILES "$<TARGET_FILE:sentry>.dSYM" DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+	elseif(MSVC)
+		sentry_install(FILES "$<$<OR:$<CONFIG:Debug>,$<CONFIG:RelWithDebInfo>>:$<TARGET_PDB_FILE:sentry>>"
+			DESTINATION "${CMAKE_INSTALL_BINDIR}")
+	endif()
 endif()
 
-if(BACKEND_CRASHPAD)
-	set(CMAKE_CXX_STANDARD 14)
+if(BUILD_SHARED_LIBS)
+	target_compile_definitions(sentry PRIVATE SENTRY_BUILD_SHARED)
 else()
-	set(CMAKE_CXX_STANDARD 11)
+	target_compile_definitions(sentry PUBLIC SENTRY_BUILD_STATIC)
 endif()
 
-#set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror=implicit-function-declaration -Werror=incompatible-function-pointer-types -Wall -fvisibility=hidden")
-if(NOT WIN32)
-	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -fvisibility=hidden")
+if(SENTRY_CURL_SUPPORT)
+	find_package(CURL REQUIRED)
+	target_compile_definitions(sentry PUBLIC SENTRY_WITH_LIBCURL_TRANSPORT)
+	target_include_directories(sentry PRIVATE ${CURL_INCLUDE_DIR})
+	# The exported sentry target must not contain any path of the build machine, therefore use generator expressions
+	# FIXME: cmake 3.12 introduced the target CURL::libcurl
+	string(REPLACE ";" "$<SEMICOLON>" GENEX_CURL_LIBRARIES "${CURL_LIBRARIES}")
+	string(REPLACE ";" "$<SEMICOLON>" GENEX_CURL_COMPILE_DEFINITIONS "${CURL_COMPILE_DEFINITIONS}")
+	target_link_libraries(sentry PRIVATE $<BUILD_INTERFACE:${GENEX_CURL_LIBRARIES}>)
+	target_compile_definitions(sentry PRIVATE $<BUILD_INTERFACE:${GENEX_CURL_COMPILE_DEFINITIONS}>)
 endif()
 
-if(BACKEND_CRASHPAD AND WIN32)
-	# Disable duplicate-define warnings, as the crashpad build defines all the
-	# "slim windows.h" defines, which sentry_boot also defines.
-	add_definitions(/wd4005)
+if(WIN32)
+	target_compile_definitions(sentry
+		PRIVATE WINVER=0x0603 NTDDI_VERSION=0x06030000
+		PUBLIC SENTRY_WITH_WINHTTP_TRANSPORT
+	)
+	target_link_libraries(sentry PRIVATE winhttp)
 endif()
 
-if(ANDROID)
-	set(WITH_LIBUNWINDSTACK TRUE)
-elseif(NOT WIN32)
-	set(WITH_LIBBACKTRACE TRUE)
+#set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror=implicit-function-declaration -Werror=incompatible-function-pointer-types -Wall -fvisibility=hidden")
+set_property(TARGET sentry PROPERTY C_VISIBILITY_PRESET hidden)
+if(MSVC)
+	if(CMAKE_SIZEOF_VOID_P EQUAL 4)
+		set(CMAKE_ASM_MASM_FLAGS "${CMAKE_ASM_MASM_FLAGS} /safeseh")
+	endif()
+else()
+	target_compile_options(sentry PUBLIC $<BUILD_INTERFACE:-Wall>)
 endif()
 
-include_directories("include")
-include_directories("src")
-
-file(GLOB_RECURSE SENTRY_NATIVE_TEST_SOURCES
-	${CMAKE_CURRENT_SOURCE_DIR}/tests/unit/*.c
+target_include_directories(sentry PUBLIC
+	"$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>"
+	"$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src>"
+	"$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>"
 )
 
+# Without this, dladdr can't find functions
+target_link_libraries(sentry INTERFACE
+	"$<$<OR:$<PLATFORM_ID:Linux>,$<PLATFORM_ID:Android>>:-Wl,--build-id=sha1,-E>")
```

Comment:
We had this only for the unit tests at first, since they need to have their functions findable by `dladdr`, but I guess this is fine to add in general

Reply:
`-E` was really intended for tests only, and I would prefer to keep it on the test target. 

We do need `--build-id=sha1` on the library in case we want to support building with `clang` on Linux. Otherwise, `clang` does not generate a build note which Sentry needs to compute a debug identifier. So, we can keep that flag here.

Reply:
So keep it as is?
This link flag will also be exported when packaging `sentry-native`.

Reply:
Maybe copy-paste the code, once without the `-E` for normal libsentry, and once with `-E` for the unit tests.

Reply:
Yup, that's what is done now.